nix/nixops.nix

```diff
@@ -136,22 +132,48 @@ in {
       pkgs.vegeta
       (
         pkgs.writeShellScriptBin "vegeta-bench" ''
+          set -euo pipefail
+
           # rate=0 means maximum rate subject to max-workers
           echo "GET http://server/pathological?status=200" | vegeta attack -rate=0 -duration=1s -max-workers=1 | tee results.bin | vegeta report
         ''
       )
       (
         pkgs.writeShellScriptBin "vegeta-bench-max-requests" ''
+          set -euo pipefail
+
           # rate=0 means maximum rate subject to max-workers
           echo "GET http://server/pathological?status=200" | vegeta attack -rate=0 -duration=10s -max-workers=50 | tee results.bin | vegeta report
         ''
       )
       (
-        pkgs.writeShellScriptBin "net-bench" ''
+        pkgs.writeShellScriptBin "psql-net-bench" ''
+          set -euo pipefail
+
+          psql -U postgres -c "TRUNCATE net._http_response; TRUNCATE net.http_request_queue;"
+          psql -U postgres -c "alter system set pg_net.batch_size to 32000;" # this just a high number
+          psql -U postgres -c "select net.worker_restart();"
           psql -U postgres -c "truncate net._http_response;"
           psql -U postgres -c "select net.http_get('http://server/pathological?status=200') from generate_series(1, 400);" > /dev/null
           sleep 2
           psql -U postgres -c "select * from pg_net_stats;"
+          psql -U postgres -c "alter system reset pg_net.batch_size;"
+          psql -U postgres -c "select net.worker_restart();"
+        ''
+      )
+      (
+        pkgs.writeShellScriptBin "psql-reproduce-timeouts" ''
+          set -euo pipefail
+
+          number_of_requests=10000
+          sleep_time=60 # Using the default batch_size (200), this is enough time to finish the whole $number_of_requests
```

Comment:
Considering the default batch_size (200), we should be able to wait ~50 seconds to get all the 10K responses. However this is not the case, most likely due to the `select()` overhead as mentioned on https://github.com/supabase/pg_net/pull/150#issuecomment-2354262406.